sdk/python/lib/test/langhost/component_dependencies/__main__.py

```diff
@@ -39,3 +39,8 @@ def __init__(self, name, opts=None):
 resH = MyResource("resH", {"propA": comp2})
 resI = MyResource("resI", {"propA": resG})
 resJ = MyResource("resJ", {}, ResourceOptions(depends_on=[comp2]))
+
+first = MyComponent("first")
+firstChild = MyResource("firstChild", {}, ResourceOptions(parent=first))
+second = MyComponent("second", ResourceOptions(parent=first, depends_on=[first]))
+myresource = MyResource("myresource", {}, ResourceOptions(parent=second))
```

Comment:
To check my understanding, the reason that `second` hangs is because it's running `second.registerResource()` and depending on its child to finish `myresource.registerResource()`, since `second` is its parent. It needs to know `second`'s URN to finish. So `second.registerResource()` was waiting on itself to finish. Which also means that `first` is hung because it's waiting on `second`.

Is this correct? If not, can you correct my understanding?

Reply:
Yep